frontend/src/metabase-lib/v1/expressions/types.ts

```diff
@@ -51,3 +51,10 @@ export type ErrorWithMessage = {
   pos?: number | null;
   len?: number | null;
 };
+
+export type Token = {
```

Comment:
Moved from `diagnostics.ts` because it's useful in `tokenizer` now that it's been converted to TS.
Made `op` optional.